src/micro_args.c

```diff
@@ -0,0 +1,356 @@
+        return;
+
+    for (i = 0; i < args->count; i++)
+    {
+        NATS_FREE(args->args[i]);
+    }
+    NATS_FREE(args->args);
+    NATS_FREE(args);
+}
+
+int microArgs_Count(microArgs *args)
+{
+    if (args == NULL)
+        return 0;
+
+    return args->count;
+}
+
+microError *
+microArgs_GetInt(int *val, microArgs *args, int index)
+{
+    if ((args == NULL) || (index < 0) || (index >= args->count) || (val == NULL))
+        return micro_ErrorInvalidArg;
+
+    *val = *((int *)args->args[index]);
+    return NULL;
+}
+
+microError *
+microArgs_GetFloat(long double *val, microArgs *args, int index)
+{
+    if ((args == NULL) || (index < 0) || (index >= args->count) || (val == NULL))
+        return micro_ErrorInvalidArg;
+
+    *val = *((long double *)args->args[index]);
+    return NULL;
+}
+
+microError *
+microArgs_GetString(const char **val, microArgs *args, int index)
+{
+    if ((args == NULL) || (index < 0) || (index >= args->count) || (val == NULL))
+        return micro_ErrorInvalidArg;
+
+    *val = (const char *)args->args[index];
+    return NULL;
+}
+
+// decodes the rest of a string into a pre-allocated buffer of sufficient
+// length, or just calculates the needed buffer size. The opening quote must
+// have been already processed by the caller (parse).
+static microError *
+decode_rest_of_string(char *dup, int *decoded_len, int *i, const char *data, int data_len)
+{
+    char c;
+    int len = 0;
+    bool terminated = false;
+    bool escape = false;
+
+    for (; !terminated && *i < data_len; (*i)++)
+    {
+        c = data[*i];
+        switch (c)
+        {
+        case '"':
+            if (escape)
+            {
+                // include the escaped quote.
+                if (dup != NULL)
+                {
+                    dup[len] = c;
+                }
+                len++;
+                escape = false;
+            }
+            else
+            {
+                // end of quoted string.
+                terminated = true;
+            }
+            break;
+
+        case '\\':
+            if (!escape)
+            {
+                escape = true;
+            }
+            else
+            {
+                // include the escaped backslash.
+                if (dup != NULL)
+                {
+                    dup[len] = c;
+                }
+                len++;
+                escape = false;
+            }
+            break;
+
+        default:
+            if (dup != NULL)
+            {
+                dup[len] = c;
+            }
+            len++;
+            escape = false;
+            break;
+        }
+    }
+    if (!terminated)
+    {
+        return micro_Errorf("a quoted string is not properly terminated");
+    }
+
+    *decoded_len = len;
+    return NULL;
+}
+
+static microError *
+decode_and_dupe_rest_of_string(char **dup, int *i, const char *data, int data_len)
+{
+    microError *err = NULL;
+    int start = *i;
+    int decoded_len;
```

Comment:
I guess you saw on Travis, that compiler has an issue with this not being initialized. I understand that you won't use the value if err is != NULL, but let's make the compiler happy.